### PR TITLE
add kubeadm

### PR DESCRIPTION
**Pull request by liangyuanpeng**
#### 1. Please confirm that you have read the document before PR submitted

- [x] Yes, I have read [THIS NOTE DOC.](https://github.com/kcl-lang/modules/blob/main/README.md) 

#### 2. Contact Information(Optional)

If it is convenient, please provide your contact information so we can reach you when processing the PR:

- **Email**:
- **HomePage**:

Add kubeadm mode. and this is the first PR, so this is not a complete module.

### CODE REVIEW

File: kubeadm/1.29/kcl.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[package]
+name = "kubeadm"
+edition = "*"
+version = "1.29+kcl0"
```
Comment (liangyuanpeng) [inline kubeadm/1.29/kcl.mod:4]: Is this a valid version number? if not ,I'm looking to iterate multiple times on kubeadm 1.29, any suggestions?

Comment (Peefy) [reply kubeadm/1.29/kcl.mod:4]: cc @zong-zhe

**Comment by Peefy** [comment]
Hello @liangyuanpeng, Community thanks for your contribution. Please sign the CLA and DCO to pass CI.

File: kubeadm/1.29/kcl.mod (hunk 1 above)
Comment (zong-zhe) [reply kubeadm/1.29/kcl.mod:4]: Hi @liangyuanpeng 😊

Is the `version` field value `1.29+kcl0` generated by `kcl` command ? We currently require that the package version number of KCL support the semantic version, if this version is generated, this should be a bug, welcome to open an issue and provide some more details.

the semantic version - https://semver.org/
raise an issue - https://github.com/kcl-lang/kpm/issues/new?assignees=&labels=type%2Fbug&projects=&template=bug-report.md

**Comment by liangyuanpeng** [comment]
> Please sign the CLA and DCO to pass CI.

no worries, I will do it after the PR is ready


@Peefy  BTW,in order to be visible in artifacthub, do I need to manually create artifacthub-pkg.yaml at https://github.com/kcl-lang/modules/tree/main/.integration/artifacthub? It seems that the documentation is https://github.com/kcl- lang/modules?tab=readme-ov-file#prepare-your-kcl-module is outdate.

**Comment by liangyuanpeng** [comment]
For the CLA, should we use https://github.com/cncf/cla ?

**Comment by Peefy** [comment]
> BTW,in order to be visible in artifacthub, do I need to manually create artifacthub-pkg.yaml

No, CI will automatically generate the files required for the artifact hub.

**Comment by Peefy** [comment]
> It seems that the documentation is https://github.com/kcl-lang/modules?tab=readme-ov-file#prepare-your-kcl-module is outdate.

ping @zong-zhe

**Comment by zong-zhe** [comment]
> It seems that the documentation is https://github.com/kcl- lang/modules?tab=readme-ov-file#prepare-your-kcl-module is outdate.

Hi @liangyuanpeng, I noted in the documentation that this file is optional, if you manually add this file, it will be the file you use, in general you do not need to manually create the file, the file will be automatically generated.

> Our strategy is that if there is a configuration file named artifacthub-pkg.yaml in the directory where your module's kcl.mod file is located, then we will use the artifacthub-pkg.yaml you provided to display the information of your module, otherwise, we will use some default information to generate the corresponding artifacthub-pkg.yaml file.

File: kubeadm/1.29/kcl.mod (hunk 1 above)
Comment (liangyuanpeng) [reply kubeadm/1.29/kcl.mod:4]: @zong-zhe 

> Is the version field value 1.29+kcl0 generated by kcl command

No, This is the version format I added myself

> the semantic version - https://semver.org/

Thanks, I have check it and it's not match, and `1.29.0+kcl0` is match, let me update it.

**Comment by liangyuanpeng** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by liangyuanpeng** [comment]
/unhold 

`kcl mod init` have not param of version.

**Comment by Peefy** [comment]
CI failed when push this module to docker.io with the invalid tag. https://github.com/kcl-lang/modules/actions/runs/8200198425/job/22426514617

**Pull request merged**